File: go/1137_N-th_Tribonacci_Number.go
var memo = map[int]int{
    1: 1,
    2: 1,
}

func tribonacci(n int) int {
    if (n == 0) {
        return n
    }
    if memo[n] == 0 {
        memo[n] = tribonacci(n - 1) + tribonacci(n - 2) + tribonacci(n - 3)
    }
    return memo[n]
}

File: ts/1137_N-th_Tribonacci_Number.ts
let memo = new Map<number, number>([
    [0, 0],
    [1, 1],
    [2, 1]
]);

function tribonacci(n: number): number {
    
    if (!memo.has(n)) {
        memo.set(n, tribonacci(n - 1) + tribonacci(n - 2) + tribonacci(n - 3));
    }

    return memo.get(n);
};

